core/storage/feedback/gae_models.py

```diff
@@ -420,6 +421,400 @@ def get_multi(cls, user_id, thread_ids):
+
+        Returns:
+            GeneralFeedbackThreadModel. The newly created FeedbackThreadModel
+                instance.
+
+        Raises:
+            Exception: A thread with the given thread ID exists already.
+        """
+        if cls.get_by_id(thread_id):
+            raise Exception('Feedback thread ID conflict on create.')
+        return cls(id=thread_id)
+
+    @classmethod
+    def get_threads(
+            cls, entity_type, entity_id, limit=feconf.DEFAULT_QUERY_LIMIT):
+        """Returns a list of threads associated with the entity, ordered
+        by their "last updated" field. The number of entities fetched is
+        limited by the `limit` argument to this method, whose default
+        value is equal to the default query limit.
+
+        Args:
+            entity_type: str. The type of the entity.
+            entity_id: str. The ID of the entity.
+            limit: int. The maximum possible number of items
+                in the returned list.
+
+        Returns:
+            list(GeneralFeedbackThreadModel). List of threads associated with
+                the entity. Doesn't include deleted entries.
+        """
+        return cls.get_all().filter(cls.entity_type == entity_type).filter(
+            cls.entity_id == entity_id).order(cls.last_updated).fetch(limit)
+
+
+class GeneralFeedbackMessageModel(base_models.BaseModel):
+    """Feedback messages. One or more of these messages make a thread.
+
+    The id of instances of this class has the form [THREAD_ID].[MESSAGE_ID]
+    """
+    # ID corresponding to an entry of FeedbackThreadModel.
+    thread_id = ndb.StringProperty(required=True, indexed=True)
+    # 0-based sequential numerical ID. Sorting by this field will create the
+    # thread in chronological order.
+    message_id = ndb.IntegerProperty(required=True, indexed=True)
+    # ID of the user who posted this message. This may be None if the feedback
+    # was given anonymously by a learner.
+    author_id = ndb.StringProperty(indexed=True)
+    # New thread status. Must exist in the first message of a thread. For the
+    # rest of the thread, should exist only when the status changes.
+    updated_status = ndb.StringProperty(choices=STATUS_CHOICES, indexed=True)
+    # New thread subject. Must exist in the first message of a thread. For the
+    # rest of the thread, should exist only when the subject changes.
+    updated_subject = ndb.StringProperty(indexed=False)
+    # Message text. Allowed not to exist (e.g. post only to update the status).
+    text = ndb.StringProperty(indexed=False)
+    # Whether the incoming message is received by email (as opposed to via
+    # the web).
+    received_via_email = ndb.BooleanProperty(default=False, indexed=True)
+    # TODO (nithesh): Overriding last_updated of base model for migrating
+    # instances from old model to the new one. To be removed after migration.
+    last_updated = ndb.DateTimeProperty()
```

Comment:
See above comments.

Reply:
I had already put a TODO. I think it is fine?